tests/PHPStan/Analyser/nsrt/property-null-after-assignment.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace PropertyNullAfterAssignmentStrictTypesDisabled;
+
+use function PHPStan\Testing\assertNativeType;
+use function PHPStan\Testing\assertType;
+
+class KeepsPropertyNonNullable {
+	private readonly int $i;
+
+	public function __construct()
+	{
+		$this->i = getIntOrNull();
+	}
+
+	public function doFoo(): void {
+		assertType('int', $this->i);
+		assertNativeType('int', $this->i);
+	}
+}
+
+function getIntOrNull(): ?int {
+	if (rand(0, 1) === 0) {
+		return null;
+	}
+	return 1;
+}
```

Comment:
main motivation was this case here, as with types remembered from constructors we otherwise get a lot of followup errors in instance methods

https://phpstan.org/r/40dfb9a5-a8e9-49c5-80e8-36affa12bda6